### PR TITLE
Telemetry PII on/off

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: MSAL/src/public/MSALTelemetry.h [hunk 1]
```diff
@@ -56,6 +56,12 @@
 + (nonnull MSALTelemetry *)sharedInstance;
 
 /*!
+ Set to YES to allow events possibly containing Personally Identifiable Information (PII) to be
+ sent to dispatcher.
```
Comment (oldalton) [inline MSAL/src/public/MSALTelemetry.h]: also add that default is NO

Comment (antrix1989) [reply MSAL/src/public/MSALTelemetry.h]: added

File: MSAL/src/telemetry/MSALTelemetry.m [hunk 2]
```diff
@@ -210,7 +199,17 @@ - (void)flush:(NSString *)requestId
...
         {
-            events = [@[[s_defaultEvent getProperties]] arrayByAddingObjectsFromArray:events];
+            events = [@[[[s_defaultEvent getProperties] mutableCopy]] arrayByAddingObjectsFromArray:events];
+        }
+        
+        // Clear PII fields.
+        for (NSMutableDictionary *eventProperties in events) {
+            for (NSString *propertyName in [eventProperties allKeys]) {
+                BOOL isPii = [MSALTelemetryPiiRules isPii:propertyName];
+                if (isPii && !self.piiEnabled) {
```
Comment (unpluggedk) [inline MSAL/src/telemetry/MSALTelemetry.m]: Why don't we check self.piiEnabled in the first place?

File: MSAL/src/telemetry/MSALTelemetry.m [hunk 3]
```diff
@@ -210,7 +199,21 @@ - (void)flush:(NSString *)requestId
         
         if ([events count])
         {
-            events = [@[[s_defaultEvent getProperties]] arrayByAddingObjectsFromArray:events];
+            events = [@[[[s_defaultEvent getProperties] mutableCopy]] arrayByAddingObjectsFromArray:events];
+        }
+        
+        if (!self.piiEnabled)
```
Comment (oldalton) [inline MSAL/src/telemetry/MSALTelemetry.m]: 👍

File: MSAL/test/unit/MSALTelemetryTests.m [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    [MSALTelemetry sharedInstance].piiEnabled = NO;
+}
+
+- (void)test_telemetryPiiRules_whenPiiEnabledNo_shouldSetPiiFieldsToEmpty
+{
+    [MSALTelemetry sharedInstance].piiEnabled = NO;
+    NSString *requestId = [[MSALTelemetry sharedInstance] telemetryRequestId];
+    NSString *eventName = @"test event";
+    MSALTelemetryDefaultEvent *event = [[MSALTelemetryDefaultEvent alloc] initWithName:eventName context:nil];
+    [event setProperty:MSAL_TELEMETRY_KEY_DEVICE_IP_ADDRESS value:@"192.168.0.1"];
```
Comment (oldalton) [inline MSAL/test/unit/MSALTelemetryTests.m]: Should we delete the IP address all together like in ADAL? In that case those tests will need to be changed.

Comment (antrix1989) [reply MSAL/test/unit/MSALTelemetryTests.m]: Deleted IP address and hashed PII fields in separate PR.

**Pull request merged**